lib/xml_utils.pm

```diff
@@ -51,26 +51,36 @@ Verifies that node by given XPath is unique and has expected value. C<%args> is
 
 =item * C<xpath> - XPath to the node which value we want to check
 
-=item * C<expected_val> - expected value for the node
+=item * C<expected_value> - expected value for the node`
+
+=item * C<search_by_value> - search is performed by value instead of by node, which is default`
 
 =back
 
 =cut
 
 sub verify_option {
     my (%args) = @_;
-
-    my @nodes = find_nodes(%args);
-    ## Verify that there is node found by xpath and it's single one
-    if (scalar @nodes != 1) {
-        return "Generated autoinst.xml contains unexpected number of nodes for xpath: $args{xpath}. Found: " . scalar @nodes . ", expected: 1.";
+    if ($args{search_by_value}) {
+        my $value = $args{xpc}->findvalue($args{xpath});
+        if ($value ne $args{expected_value}) {
+            return "Unexpected value for xpath $args{xpath}. Expected: '$args{expected_value}', got: '$value'";
+        }
     }
-    if ($nodes[0]->to_literal ne $args{expected_val}) {
-        return "Unexpected value for xpath $args{xpath}. Expected: '$args{expected_val}', got: '$nodes[0]'";
+    else {
+        my @nodes = find_nodes(%args);
+        if (scalar @nodes == 0) {
+            return "Expected xpath $args{xpath} didn't return any node";
+        }
+        # Verify that there is node found by xpath and it's single one
+        if (scalar @nodes != 1) {
+            return "Generated autoinst.xml contains unexpected number of nodes for xpath: $args{xpath}. Found: " . scalar @nodes . ", expected: 1.";
+        }
+        if ($nodes[0]->to_literal ne $args{expected_value}) {
+            return "Unexpected value for xpath $args{xpath}. Expected: '$args{expected_value}', got: '$nodes[0]'";
+        }
```

Comment:
And here same here? https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/9058/files#r353295084

Reply:
In this case is not even recommended, because we have more than one condition...and the postfix notation is even less clear.